sonar-batch/src/main/java/org/sonar/batch/rule/RuleFinderCompatibility.java

```diff
@@ -26,8 +31,18 @@
 import org.sonar.api.rules.RuleFinder;
 import org.sonar.api.rules.RuleQuery;
 
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
+/**
+ * FIXME Waiting for the list of all server rules on batch side this is implemented by redirecting on ActiveRules. This is not correct
+ * since there is a difference between a rule that doesn't exists and a rule that is not activated in project quality profile.
+ *
+ */
 public class RuleFinderCompatibility implements RuleFinder {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)"RuleFinder" is deprecated, implement the suggested replacement instead.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1724)